### PR TITLE
Expose secure string

### PR DESCRIPTION
**Pull request by AlexGhiondea**

### CODE REVIEW

File: src/System.Security.Cryptography.Primitives/src/System.Security.Cryptography.Primitives.TypeForwards.cs [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+[assembly: System.Runtime.CompilerServices.TypeForwardedTo(typeof(System.Security.Cryptography.CryptographicException))]
```
Comment (weshaggard) [inline src/System.Security.Cryptography.Primitives/src/System.Security.Cryptography.Primitives.TypeForwards.cs]: This shouldn't be needed. It should get copied from the ref (assuming it is a partial facade).

File: src/System.Security.SecureString/ref/System.Security.SecureString.cs [hunk 2]
```diff
@@ -5,29 +5,15 @@
...
-        public void SetAt(int index, char c) { }
-    }
-    public static partial class SecureStringMarshal
-    {
-        public static System.IntPtr SecureStringToCoTaskMemAnsi(System.Security.SecureString s) { throw null; }
-        public static System.IntPtr SecureStringToCoTaskMemUnicode(System.Security.SecureString s) { throw null; }
-        public static System.IntPtr SecureStringToGlobalAllocAnsi(System.Security.SecureString s) { throw null; }
-        public static System.IntPtr SecureStringToGlobalAllocUnicode(System.Security.SecureString s) { throw null; }
+    [System.CLSCompliant(false)]
+    public static class SecureStringMarshal {
```
Comment (weshaggard) [inline src/System.Security.SecureString/ref/System.Security.SecureString.cs]: We should consider moving this into interop services as well.

**Comment by weshaggard** [comment]
FYI @yizhang82

File: src/System.Security.SecureString/src/System/Security/SecureStringMarshal.CoreCLR.cs [hunk 3]
```diff
@@ -4,14 +4,12 @@
...
-
-        public static IntPtr SecureStringToGlobalAllocAnsi(SecureString s) => s != null ? s.MarshalToString(globalAlloc: true, unicode: false) : IntPtr.Zero;
-
-        public static IntPtr SecureStringToCoTaskMemUnicode(SecureString s) => s != null ? s.MarshalToString(globalAlloc: false, unicode: true) : IntPtr.Zero;
-
-        public static IntPtr SecureStringToGlobalAllocUnicode(SecureString s) => s != null ? s.MarshalToString(globalAlloc: true, unicode: true) : IntPtr.Zero;
+        public static IntPtr SecureStringToCoTaskMemAnsi(SecureString s) => IntPtr.Zero;
+        public static IntPtr SecureStringToGlobalAllocAnsi(SecureString s) => IntPtr.Zero;
+        public static IntPtr SecureStringToCoTaskMemUnicode(SecureString s) => IntPtr.Zero;
+        public static IntPtr SecureStringToGlobalAllocUnicode(SecureString s) => IntPtr.Zero;
```
Comment (stephentoub) [inline src/System.Security.SecureString/src/System/Security/SecureStringMarshal.CoreCLR.cs]: Why are all of these now returning IntPtr.Zero?

Comment (AlexGhiondea) [inline src/System.Security.SecureString/src/System/Security/SecureStringMarshal.CoreCLR.cs]: SecureString is now implemented in CoreLib and the methods that would do that conversion are internal.

They are exposed on the Marshal class with this PR: https://github.com/dotnet/coreclr/pull/7985 and there is an issue to move these APIs to use the Marshal APIs once the new CoreCLR is ready.https://github.com/dotnet/corefx/issues/13331

**Pull request merged**